dramatiq/worker.py

```diff
@@ -492,15 +495,13 @@ def process_message(self, message):
 
             self.broker.emit_after("process_message", message, result=res)
 
-        except SkipMessage:
+        except SkipMessage as e:
+            message.fail()
```

Comment:
`SkipMessage`s shouldn't be marked as having failed. The idea behind `SkipMessage` is to let users ack messages that they don't intend to process. Nacking means messages get put on the DLQ, which is semantically the wrong thing here.

Reply:
Ok that makes sense. Thanks for the feedback. But what would you expect the stored result to be if a user implements a middleware that skips messages, but does not fail them? Would it be `None` or a `SkipMessage` exception? If it should be a `SkipMessage` exception then I can work this in by adding an `after_skip_message` implementation to the results middleware that is similar to its `after_nack` implementation, but checks for a non-None `message._exception` (rather than `message.failed`) to store the exception.

Reply:
I think the result of a skipped message should probably be `None`.

Adding `after_skip_message` to the `Results` middleware conflicts with the `after_nack` hook (well, it doesn't exactly "conflict", but it's confusing): the `AgeLimit` middleware both fails the message and raises a Skip, so both hooks run, and it happens to be that `after_skip` runs first, then `after_nack`, so things work out OK in that case. 

More on `None` vs exception: neither option is ideal, but my feeling is it's more natural for a skipped message's result to be `None` than it is for it to be an exception. Both are equally bad in code that doesn't expect them, so we should document this interaction between `AgeLimit` and `Results`. In code that expects actors to potentially skip messages, dealing with `None` is simpler.

Reply:
Ok that makes sense. I agree that neither option is ideal. I have put up another commit addressing this.